File: js/managers/configManager.js
const fs = require('fs');
const path = require('path');
require('dotenv').config({ path: path.resolve(__dirname, '../../.env') });

class ConfigManager {
    /**
     * Handles loading, updating, and saving configuration for scrapers.
     */
    constructor(platform) {
        this.configPath = path.resolve(__dirname,'../../' ,process.env.CONFIG_PATH);
        if (!this.configPath || !fs.existsSync(this.configPath)) {
            throw new Error(`Config Path not set or file does not exist: ${this.configPath}`);
        }
        this.config = this.loadConfig(platform);
    }

    loadConfig(platform) {
        /**
         * Load the config.json into memory.
         */
        const raw = fs.readFileSync(this.configPath, 'utf-8');
        const config = JSON.parse(raw);

        // Ensure platform section exists
        const platformConfig = config[platform] || {};

        // Override with environment variables if present
        platformConfig.email = process.env[`${platform.toUpperCase()}_EMAIL`];
        platformConfig.password = process.env[`${platform.toUpperCase()}_PASSWORD`];

        // Put the updated platform config back
        config[platform] = platformConfig;

        return config;
    }

    getPlatformConfig(platformName) {
        /**
         * Get configuration for a specific platform (e.g., linkedin, naukri).
         */
        return this.config[platformName] || {};
    }

    getTitlesDict() {
        /**
         * Get the dictionary of job titles (enabled/disabled) for all platforms.
         */
        return this.config.titles || {};
    }

    updateTitle(title, enabled = false) {
        /**
         * Add a new job title or update existing one.
         */
        if (!this.config.titles) {
            this.config.titles = {};
        }
        this.config.titles[title] = enabled;
    }

    saveConfig() {
        /**
         * Write the current config object back to config.json.
         */
        fs.writeFileSync(this.configPath, JSON.stringify(this.config, null, 2));
        console.log(`✅ Config saved to ${this.configPath}`);
    }
}

module.exports = ConfigManager;

File: python/managers/config_manager.py
import json
import os
from dotenv import load_dotenv


class ConfigManager:
    """
    Handles loading, updating, and saving configuration for scrapers.
    """

    def __init__(self, platform):
        load_dotenv()
        self.config_path = os.getenv("CONFIG_PATH")
        
        if not self.config_path or not os.path.exists(self.config_path):
            raise FileNotFoundError(f"Config Path not set or file does not exist: {self.config_path}")
        
        self.config = self.load_config(platform)

     
    def load_config(self, platform):
        """
        Load the config.json into memory.
        """
        with open(self.config_path, "r") as f:
            config = json.load(f)

        # Ensure platform section exists
        platform_config = config.get(platform, {})

        # Override with environment variables if present
        platform_config["email"] = os.getenv(f"{platform.upper()}_EMAIL", platform_config.get("email", ""))
        platform_config["password"] = os.getenv(f"{platform.upper()}_PASSWORD", platform_config.get("password", ""))

        # Put the updated platform config back
        config[platform] = platform_config

        return config

    def get_platform_config(self, platform_name):
        """
        Get configuration for a specific platform (e.g., linkedin, naukri).
        """
        return self.config.get(platform_name, {})

    def get_titles_dict(self):
        """
        Get the dictionary of job titles (enabled/disabled) for all platforms.
        """
        return self.config.get("titles", {})

    def update_title(self, title, enabled=False):
        """
        Add a new job title or update existing one.
        """
        if "titles" not in self.config:
            self.config["titles"] = {}
        self.config["titles"][title] = enabled

    def save_config(self):
        """
        Write the current config object back to config.json.
        """
        with open(self.config_path, "w") as f:
            json.dump(self.config, f, indent=2)
        print(f"✅ Config saved to {self.config_path}")
